tests/integration-tests/tests/api/test_api_client.py

```diff
@@ -136,3 +143,88 @@ def _test_delete_cluster(client, cluster, region):
     response = client.list_clusters(region=region)
     cluster_names = [c["cluster_name"] for c in response["items"]]
     assert_that(cluster_names).does_not_contain(cluster_name)
+
+
+def test_official_images(region, api_client):
+    client = image_operations_api.ImageOperationsApi(api_client)
+    response = client.describe_official_images(region=region)
+    assert_that(response["items"]).is_not_empty()
+
+
+@pytest.mark.usefixtures("instance")
+def test_custom_image(request, region, os, pcluster_config_reader, api_client):
+    base_ami = retrieve_latest_ami(region, os)
+
+    config_file = pcluster_config_reader(config_file="image.config.yaml", parent_image=base_ami)
+    with open(config_file) as config_file:
+        config = config_file.read()
+
+    stack_name = generate_stack_name("integ-tests-build-image", request.config.getoption("stackname_suffix"))
+    client = image_operations_api.ImageOperationsApi(api_client)
+
+    with soft_assertions():
+        resp = _test_build_image(client, config, region, stack_name)
+        LOGGER.info("Build image response: %s", resp)
+        image = resp["image"]
+        image_id = image["image_id"]
+        assert_that(image_id).is_equal_to(stack_name)
+
+        _test_describe_image(client, image, region)
+        _test_list_images(client, image, region, "PENDING")
+
+        # CFN stack is deleted as soon as image is available
+        _cloudformation_wait(region, stack_name, "stack_delete_complete")
```

Comment:
isn't this going to timeout?

Reply:
Not sure about what you mean. [Waiters](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#waiters) will poll a resource until it reaches the desired state or an error occurs.

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Waiter.StackDeleteComplete

Polls CloudFormation.Client.describe_stacks() every 30 seconds until a successful state is reached. An error is returned after 120 failed checks.

according to that it timeouts after 1hour. it can be increased